### PR TITLE
[#156913774,#157151312] ask proceed with transaction

### PR DESCRIPTION
**Pull request by valentinacala**
The updates on Walletstyles would be used for the further style review of the screens related to the add payment method procedure. 

TODO: include preview of the selected payment method

whatsapp_image_2018-06-08_at_11 37 23 4
whatsapp_image_2018-06-08_at_11 37 42

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#156913774](https://www.pivotaltracker.com/story/show/156913774): Come CIT voglio visualizzare una preview del pagamento che sto per effettuare prima di procedere con il pagamento stesso

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: locales/en.json [hunk 1]
```diff
@@ -141,6 +141,18 @@
       "addButton": "Add payment method",
       "addDescription": "You can add your payment methods (e.g. credit cards, bank accounts) so that you don't have to re-add them from scratch for every transaction"
     },
+    "ConfirmPayment": {
+      "header":"Confirm transaction",
+      "cancel":"CANCEL",
+      "askConfirm":"Want to proceed with the transaction?",
```
Comment (fgiobergia) [inline locales/en.json]: (Do you) want to...

File: locales/en.json [hunk 2]
```diff
@@ -141,6 +141,18 @@
...
+      "partialAmount":"Amount to pay",
+      "fee":"Fee",
+      "why":"Why?",
+      "totalAmount":"Total amount",
+      "info":"Warning: after this step it will not be possile to come back",
```
Comment (fgiobergia) [inline locales/en.json]: (typo) possile -> possible

File: locales/it.json [hunk 3]
```diff
@@ -139,6 +139,18 @@
       "addButton": "Aggiungi un metodo di pagamento",
       "addDescription": "Puoi aggiungere i tuoi metodi di pagamento (carte, conti correnti, ecc.) per non doverli inserire ad ogni transazione"
     },
+    "ConfirmPayment": {
+      "header":"Conferma pagamento",
+      "cancel":"ANNULLA",
+      "askConfirm":"Confermi il pagamento?",
+      "partialAmount":"Importo da pagare",
+      "fee":"Costi di commissione",
+      "why":"Perhè?",
```
Comment (fgiobergia) [inline locales/it.json]: (typo) Perhè -> Perché

File: locales/it.json [hunk 4]
```diff
@@ -139,6 +139,18 @@
...
+      "totalAmount":"Totale",
+      "info":"Attenzione: dopo queto passaggio non sarà più possibile annullare",
```
Comment (fgiobergia) [inline locales/it.json]: (typo) queto -> questo

File: ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  }
+
+  public render(): React.ReactNode {
+    const IMPORTO = "€ 199,00";
+    const FEE = "€ 0,50";
+    const TOTALEALL = "€ 199,50";
+
+    return (
+      <Container>
+        <AppHeader>
```
Comment (fgiobergia) [inline ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx]: Since this kind of AppHeader ((optional) back button + page title) we should consider migrating it to its own component, and then use it as follows
```
<StandardAppHeader allowBack={true} title={title} />
```
It might be worth opening a story to address this and link it here

File: ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+          </Body>
+        </AppHeader>
+
+        <Content original={true}>
+          <PaymentBannerComponent navigation={this.props.navigation} />
+          <View style={WalletStyles.paddedLR}>
+            <View spacer={true} large={true} />
+            <H1>{I18n.t("wallet.ConfirmPayment.askConfirm")}</H1>
+            <View spacer={true} large={true} />
+            <Text>CREDIT CARD COMPONENT</Text>
```
Comment (fgiobergia) [inline ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx]: Link to pivotal story

File: ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+                  <Text bold={true} style={WalletStyles.textRight}>
+                    {IMPORTO}
+                  </Text>
+                </Col>
+              </Row>
+              <Row>
+                <Col size={4}>
+                  <Text>
+                    {I18n.t("wallet.ConfirmPayment.fee")}
+                    <Text>{""} </Text>
```
Comment (fgiobergia) [inline ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx]: Already mentioned somewhere else by @cloudify i believe.
I would personally recommend using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for the purpose:
```
<Text>{`${I18n.t("wallet.ConfirmPayment.fee")} `}</Text>
```

File: locales/en.json (hunk 1 above)
Comment (cloudify) [reply locales/en.json]: perhaps just "Proceed with the transaction?" would be ok, anyway we'll ask @gvarisco to review the English copy before releasing :D

File: ts/components/wallet/PaymentBannerComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import I18n from "../../i18n";
+import { WalletStyles } from "../styles/wallet";
+
+type Props = Readonly<{
+  navigation: NavigationScreenProp<NavigationState>;
+}>;
+
+export default class PaymentBannerComponent extends React.Component<Props> {
+  public render(): React.ReactNode {
+    const NOMEAVVISO = "Tari 2018";
```
Comment (cloudify) [inline ts/components/wallet/PaymentBannerComponent.tsx]: I suggest to define a specific type of this kind of content, as we are going to provide the same data to multiple components is better to share the object

File: ts/navigation/WalletNavigator.ts [hunk 9]
```diff
@@ -1,11 +1,15 @@
...
 import { AddPaymentMethodScreen } from "../screens/wallet/AddPaymentMethodScreen";
+import { ConfirmToProceedTransactionScreen } from "../screens/wallet/ConfirmToProceedTransactionScreen";
 import { CreditCardsScreen } from "../screens/wallet/CreditCardsScreen";
 import { TransactionDetailsScreen } from "../screens/wallet/TransactionDetailsScreen";
 import { TransactionsScreen } from "../screens/wallet/TransactionsScreen";
 import { WalletHomeScreen } from "../screens/wallet/WalletHomeScreen";
 import ROUTES from "./routes";
 
+/**
+ * TODO: migrate WALLET_TRANSACTION_SUMMARY on a new navigator for screens which does not visualize the footer menu.
```
Comment (cloudify) [inline ts/navigation/WalletNavigator.ts]: if this TODO is not part of this PR, create a story and link to it here

File: ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,153 @@
+/**
+ * This screen ask the authorization to proceed with the transaction.
+ * TODO: integrate credit card componet for visualization of the payment method selected for the transaction
```
Comment (cloudify) [inline ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx]: if this TODO is not part of this PR, create a story and link to it here

File: ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+  constructor(props: Props) {
+    super(props);
+  }
+
+  private goBack() {
+    this.props.navigation.goBack();
+  }
+
+  public render(): React.ReactNode {
+    const IMPORTO = "€ 199,00";
```
Comment (cloudify) [inline ts/screens/wallet/ConfirmToProceedTransactionScreen.tsx]: define a type -> create a mock value outside this component + link to the story that will implement the real data-> pass the mock value to the props of this component

File: ts/theme/components/Content.ts [hunk 12]
```diff
@@ -5,6 +5,7 @@ declare module "native-base" {
   namespace NativeBase {
     interface Content {
       alternative?: boolean;
+      original?: boolean;
```
Comment (fpersico) [inline ts/theme/components/Content.ts]: What you think about 'nopadding` or something similar?

**Comment by valentinacala** [comment]
paywith
paywith2

paywith 3

File: ts/components/wallet/PaymentBannerComponent.tsx [hunk 13]
```diff
@@ -0,0 +1,57 @@
+/**
+ * This component displays a summary on the transaction.
+ * Used for the screens from the identification of the transaction to the end of the procedure.
+ * TODO: integrate with walletAPI
```
Comment (cloudify) [inline ts/components/wallet/PaymentBannerComponent.tsx]: link to relevant story

File: ts/components/wallet/payWith/addNewPaymentMethodComponent.tsx [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+import { Grid, Row } from "react-native-easy-grid";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import I18n from "../../../i18n";
+import ROUTES from "../../../navigation/routes";
+import Icon from "../../../theme/font-icons/io-icon-font/index";
+import variables from "../../../theme/variables";
+
+type Props = Readonly<{
+  navigation: NavigationScreenProp<NavigationState>;
+  isInTransaction?: boolean;
```
Comment (cloudify) [inline ts/components/wallet/payWith/addNewPaymentMethodComponent.tsx]: it looks like this property is never read, also why it's optional? it's either inside a transaction or it's not

File: ts/screens/wallet/ChoosePaymentMethodScreen.tsx [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+          <View style={WalletStyles.paddedLR}>
+            <View spacer={true} />
+            <H1> {I18n.t("wallet.payWith.title")} </H1>
+
+            <View spacer={true} />
+            <Text> {I18n.t("wallet.payWith.info")}</Text>
+            <View spacer={true} />
+            <List
+              removeClippedSubviews={false}
+								dataArray={cards as any[]} // tslint:disable-line
```
Comment (cloudify) [inline ts/screens/wallet/ChoosePaymentMethodScreen.tsx]: explicitly disable rules

File: ts/types/wallet.ts [hunk 16]
```diff
@@ -33,3 +33,16 @@ export const UNKNOWN_TRANSACTION: WalletTransaction = {
   transactionCost: 0,
   isNew: false
 };
+
+/**
+ * This type represent the details of a transaction the user should display
+ * to see summary of the transaction.
+ */
+
+export type TransactionSummary = {
```
Comment (cloudify) [inline ts/types/wallet.ts]: wrap with `Readonly<>`

File: ts/components/wallet/payWith/addNewPaymentMethodComponent.tsx [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+}>;
+
+type Route = keyof typeof ROUTES;
+
+interface IPaymentMethod {
```
Comment (cloudify) [inline ts/components/wallet/payWith/addNewPaymentMethodComponent.tsx]: either convert to a type and wrap with `Readonly<>` or mark all properties as `readonly`

File: locales/en.json [hunk 18]
```diff
@@ -152,8 +152,28 @@
...
+      "cancel":"CANCEL",
+      "cancelPayment":"Cancel the transaction",
+      "goToPay":"Confirm and pay",
+      "askConfirm":"Want to proceed with the transaction?",
+      "partialAmount":"Amount to pay",
+      "fee":"Fee",
+      "why":"Why?",
+      "totalAmount":"Total amount",
+      "info":"Warning: after this step it will not be possible to come back",
+      "info2":" You are using your favourite payment method.",
```
Comment (cloudify) [inline locales/en.json]: remove the extra space at the beginning

File: locales/it.json [hunk 19]
```diff
@@ -150,8 +150,28 @@
...
     },
+    "ConfirmPayment": {
+      "header":"Conferma pagamento",
+      "cancel":"ANNULLA",
+      "cancelPayment":"Annulla il pagamento",
+      "goToPay":"Conferma e paga",
+      "askConfirm":"Confermi il pagamento?",
+      "partialAmount":"Importo da pagare",
+      "fee":"Costi di commissione",
+      "why":"Perhè?",
```
Comment (cloudify) [inline locales/it.json]: typo

File: ts/components/wallet/payWith/addNewPaymentMethodComponent.tsx (hunk 14 above)
Comment (cloudify) [reply ts/components/wallet/payWith/addNewPaymentMethodComponent.tsx]: @valentinacala remove the question mark
